### ISSUE TITLE
Problem with "Walkthrough: Create an n-tier data application"

### ISSUE THREAD

**Issue by BentJes** [MicrosoftDocs/visualstudio-docs#7340]
I started using Visual Basic but could not get the following C# code translated to Visual Basic
"   [OperationContract]
     DataEntityTier.NorthwindDataSet.CustomersDataTable GetCustomers();

     [OperationContract]
     DataEntityTier.NorthwindDataSet.OrdersDataTable GetOrders();"

in working order. and got stopped.

Went to C# and I have almost finished the walkthrough, but got stopped at the very end at "Increase the maximum message size allowed by the service".
It was not possible to locate the maxReceivedMessage size attribute by double-clicking the App.config file in the PresentationTier project and without the project could not run.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: bfe690a0-e17d-4a9c-a107-7189e29fe815
* Version Independent ID: 87bbbd28-a5a0-76dd-468d-52c4a930f8cd
* Content: [Walkthrough: Creating an N-tier Data Application - Visual Studio (Windows)](https://docs.microsoft.com/en-us/visualstudio/data-tools/walkthrough-creating-an-n-tier-data-application?view=vs-2022)
* Content Source: [docs/data-tools/walkthrough-creating-an-n-tier-data-application.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/data-tools/walkthrough-creating-an-n-tier-data-application.md)
* Product: **visual-studio-windows**
* Technology: **vs-data-tools**
* GitHub Login: @ghogen
* Microsoft Alias: **ghogen**

**Comment by ghogen** [comment]
@BentJes You can switch the code language to VB using the language switcher at the top of the page on the left side. I know it's not very discoverable. The docs platform design team is working on improving the design. When you switch to Visual Basic, you see the following version of that code:

```
<OperationContract()> _
Function GetCustomers() As DataEntityTier.NorthwindDataSet.CustomersDataTable

<OperationContract()> _
Function GetOrders() As DataEntityTier.NorthwindDataSet.OrdersDataTable
```

 We'll test the scenario using the latest product, VS 2022, to see whether everything still works correctly, or if we need to update something.

**Comment by BentJes** [comment]
Thank you for your rapid reply.
I wil try your suggestions i in Visual Basic, but still have  the question about ”maxReceivedMessageSize”.

I have for many years extensibly used ”Office XP Access” forms as SQL_database frontend, but I would like to swift to Visual Studio.

I still have some comments on the Walkthrough.
1.   In: ” To create the n-tier solution and DataEntityTier class library”, it should be stated that Class Library (.net framework) is to be used. When using .net 6.0, the DataSourse window is not available.
a. Choosing Class Library (.net framework) invokes .net Framework 4.7.2. I is necessary to use the same .net Framework in ” DataAccessTier”, ”DataEntityTier”, ”DataService” and ”DataService”.
2.    On create the dataset:
a. The Data menu does not exist in verion 2022. Use menu View -> Other windows -> Data source -> Add New Data Source.

Med venlig hilsen
Bent Jessen

Fra: Gordon Hogenson
Sendt: 22. november 2021 22:01
Til: MicrosoftDocs/visualstudio-docs
Cc: BentJes; Mention
Emne: Re: [MicrosoftDocs/visualstudio-docs] Problem with "Walkthrough:Create an n-tier data application" (Issue #7340)

@BentJes You can switch the code language to VB using the language switcher at the top of the page on the left side. I know it's not very discoverable. The docs platform design team is working on improving the design. When you switch to Visual Basic, you see the following version of that code:
<OperationContract()> _
Function GetCustomers() As DataEntityTier.NorthwindDataSet.CustomersDataTable

<OperationContract()> _
Function GetOrders() As DataEntityTier.NorthwindDataSet.OrdersDataTable
We'll test the scenario using the latest product, VS 2022, to see whether everything still works correctly, or if we need to update something.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android. 



-- 
Denne e-mail blev kontrolleret for virusser af Avast antivirussoftware.
https://www.avast.com/antivirus

**Comment by BentJes** [comment]
Your are right about changing code language, but it is on the top right side.

Med venlig hilsen
Bent Jessen

Fra: Gordon Hogenson
Sendt: 22. november 2021 22:01
Til: MicrosoftDocs/visualstudio-docs
Cc: BentJes; Mention
Emne: Re: [MicrosoftDocs/visualstudio-docs] Problem with "Walkthrough:Create an n-tier data application" (Issue #7340)

@BentJes You can switch the code language to VB using the language switcher at the top of the page on the left side. I know it's not very discoverable. The docs platform design team is working on improving the design. When you switch to Visual Basic, you see the following version of that code:
<OperationContract()> _
Function GetCustomers() As DataEntityTier.NorthwindDataSet.CustomersDataTable

<OperationContract()> _
Function GetOrders() As DataEntityTier.NorthwindDataSet.OrdersDataTable
We'll test the scenario using the latest product, VS 2022, to see whether everything still works correctly, or if we need to update something.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android. 



-- 
Denne e-mail blev kontrolleret for virusser af Avast antivirussoftware.
https://www.avast.com/antivirus

**Comment by BentJes** [comment]
I have now tried the walkthrough: "Create an n-tier data application" in both Visual Basic and C#. In both trials I stopped at the very end at "Increase the maximum message size allowed by the service".

The doc states:
1.   In Solution Explorer, double-click the app.config file in the PresentationTier project.
2   Locate the maxReceivedMessage size attribute and change the value to 6553600.

Double-clicking the App.config file brings up an XML-file with no mention of the "maxReceivedMessage" property.

Is it possible to code in the attribute?

**Comment by ghogen** [comment]
@Bentjes
In the case that the attribute is not present, you can add it. Try something like the following:

```
<system.serviceModel>
    <bindings>
        <basicHttpBinding>
            <binding maxBufferSize="6553600" maxReceivedMessageSize="6553600" />
        </basicHttpBinding>
    </bindings>
</system.serviceModel>
```

**Comment by ghogen** [comment]
@BentJes I would appreciate it if you could confirm that the previous example addresses your concern.

**Comment by BentJes** [comment]
Thank you for your reply. Now I should bee able to continue.

Med venlig hilsen
Bent Jessen

Fra: PRMerger5
Sendt: 1. december 2021 22:52
Til: MicrosoftDocs/visualstudio-docs
Cc: BentJes; Mention
Emne: Re: [MicrosoftDocs/visualstudio-docs] Problem with "Walkthrough:Create an n-tier data application" (Issue #7340)

Closed #7340.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android. 



-- 
Denne e-mail blev kontrolleret for virusser af Avast antivirussoftware.
https://www.avast.com/antivirus
